libraries/node-core-library/src/FileSystem.ts

```diff
@@ -1246,6 +1236,97 @@ export class FileSystem {
     );
   }
 
+  private static _handleLink(linkFn: () => void, options: IInternalFileSystemCreateLinkOptions): void {
+    try {
+      linkFn();
+    } catch (error) {
+      if (FileSystem.isExistError(error)) {
+        // Link exists, handle it
+        switch (options.alreadyExistsBehavior) {
+          case AlreadyExistsBehavior.Ignore:
+            break;
+          case AlreadyExistsBehavior.Overwrite:
+            // fsx.linkSync does not allow overwriting so we must manually delete.
+            // We don't know if it is a file or a folder, so check first. We also
+            // want to use getLinkStatistics because getStatistics does not work
+            // for symlinks.
+            const stats: fs.Stats = this.getLinkStatistics(options.newLinkPath);
+            if (stats.isDirectory()) {
+              this.deleteFolder(options.newLinkPath);
+            } else {
+              this.deleteFile(options.newLinkPath);
+            }
+            linkFn();
+            break;
+          case AlreadyExistsBehavior.Error:
+          default:
+            throw error;
+        }
+      } else {
+        // When attempting to create a link in a directory that does not exist, an ENOENT
+        // error is thrown, so we should ensure the directory exists before retrying. There
+        // are also cases where the target file must exist, so validate in those cases to
+        // avoid confusing the missing directory with the missing target file.
+        if (
+          FileSystem.isNotExistError(error) &&
+          (!options.linkTargetPathMustExist || FileSystem.exists(options.linkTargetPath))
+        ) {
+          this.ensureFolder(nodeJsPath.dirname(options.newLinkPath));
+          linkFn();
+        } else {
+          throw error;
+        }
+      }
+    }
+  }
+
+  private static async _handleLinkAsync(
+    linkFn: () => Promise<void>,
+    options: IInternalFileSystemCreateLinkOptions
+  ): Promise<void> {
+    try {
+      await linkFn();
+    } catch (error) {
+      if (FileSystem.isExistError(error)) {
+        // Link exists, handle it
+        switch (options.alreadyExistsBehavior) {
+          case AlreadyExistsBehavior.Ignore:
+            break;
+          case AlreadyExistsBehavior.Overwrite:
+            // fsx.linkSync does not allow overwriting so we must manually delete.
+            // We don't know if it is a file or a folder, so check first. We also
```

Comment:
You're using `getLinkStatistics` because you are concerned with the object in the folder, not its target.